### DISCUSSION TITLE
Support for animated GIF's

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#6947] [Q&A]
I have added Animated.Gif to my project. I wonder whether there is still the plan to integrate this into the Avalonia package?

**Comment by jmacato** [comment]
> I have added Animated.Gif to my project. I wonder whether there is still the plan to integrate this into the Avalonia package?

No, i don't think we will add this to the main repo. It's not too essential enough and the repo already has alot of controls that we have to maintain indefinitely.

**Reply by robloo** [reply]
> the repo already has alot of controls that we have to maintain indefinitely.

I see variations of this comment consistently from core maintainers. As the community keeps growing I would like to see more acceptance of pulling in other controls and features. For example, anything that exists today in WPF or WinUI I would consider fair game to bring into mainline Avalonia itself for feature parity. It's true WPF never supported animations; however, WinUI has things like `AnimatedIcon` and Lottie support (that I know you are working on yourself). 

Keep in mind that as the project grows it actually stabilizes. Fundamental/core features won't be changing nearly as much and with lots of users the bugs are found quicker and fixed -- maintenance becomes easier over time. This is especially true considering the fundamental architecture of Avalonia compared to Maui or Uno. Where these other frameworks are constantly chasing platform changes Avalonia is far more stable since it just renders itself.

Add in a growing Avalonia developer community that contributes more and the situation improves even more. I don't like to hear "we can't accept new features there is too much to maintain already." Of course it's true to a degree, and I know you all have done so much, but this thinking artificially restricts innovation at a time the project is really taking off. It's a mindset I think we need to do away with.

**Comment by llfab** [comment]
Understood. What would be your preferred solution for animated gif integration in an Avalonia App (Windows and Linux)?

**Reply by jmacato** [reply]
This library works anywhere Avalonia works :) It's pure C# so despite the lack of maintenance lately, this is still the best one for the job.
